### DISCUSSION TITLE
"ERROR: update failed" when running `west update`

### DISCUSSION THREAD

**Discussion by erwango** [zephyrproject-rtos/zephyr#28668] [General] [answered]
I often get the following error when running `west update`
`ERROR: update failed for project hal_stm32`
ERROR could be reported for whatever module.

It usually happens when I have non-commited changes, but it could also happen in other cases I've not been able to characterized.

Is there a way to force `west update`  to avoid these errors?

**Comment by mbolivar-nordic** [comment, marked as answer]
Make sure to stash or commit any unstaged changes. West update uses git checkout --detach internally to update the project without overwriting your local branches, which can fail if the checkout would overwrite local uncommitted work. In this case this is west trying to help you from deleting work.

**Comment by erwango** [comment]
What if I have no file checked out ?
Here is the last occurrence:
`ERROR: update failed for project segger`
I swear I never messed up there

EDIT: I indeed have a file modified there. Not sure how it happened.
